### PR TITLE
updated use of importPaths for v2

### PR DESCRIPTION
**Pull request by Laveklint**
Hi @Zoramite,
I agree with closing the previous PR.

I have now updated to reflect your recent changes.
It still works with my previous thoughts despite the updates of Less.

I added ```paths``` to the render options as well as renamed _parserPaths_ to _importPaths_ which I felt reflected the method better.

I also changed the checkImports-method to not use imported.path since path is _imported_
I'm open for discussion J

### CODE REVIEW

File: lib/middleware.js [hunk 1]
```diff
@@ -33,10 +33,10 @@ var checkImports = function(path, next) {
   var changed = [];
 
   nodes.forEach(function(imported){
-    fs.stat(imported.path, function(err, stat) {
```
Comment (Zoramite) [inline lib/middleware.js]: I need to do some more testing, but I think I may have broke something the `imported.path` change, so I am going to hold off on this until I test it.

**Comment by Zoramite** [comment]
Alright, I modified things a bit. The way that you had it could have caused race conditions where the overall options are being changed per request but another request could have changed the value before it was used and cause some weird errors and side effects.

Also fixed a similar condition that already existed with the filename, so that is good as well.

File: lib/middleware.js (hunk 1 above)
Comment (Laveklint) [inline lib/middleware.js]: If the method is run it breaks with the ```imported.path```, so yes, something is not quite right there. Did you previously map each import to an object with had a path-key? I'm keeping this in the fork for now but I'll see if I can find some time to have a look at it.

**Comment by Laveklint** [comment]
Good catch. 
Thanks for the merge.

**Pull request merged**